### PR TITLE
Add unit tests for objects.go

### PR DESCRIPTION
**Pull request by atoato88**
This PR adds some unit tests for `objects.go` because that has some core functions for this framework.
It's good to increase coverage for framework.

This PR doesn't have enough coverage, so I'll update to add more unit tests for `objects.go`.

### CODE REVIEW

**Comment by atoato88** [comment]
/assign @justinsb 
Could you review this?

I've implements tests for objects.go
I think this PR covers core functions (not all functions).
I may change some test code for more effective and ease to read.

File: pkg/patterns/declarative/pkg/manifest/objects_test.go [hunk 1]
```diff
@@ -151,3 +154,831 @@ configMapGenerator:
...
+				t.Fatalf("unexpected error: %v", err)
+			}
+			for _, o := range objects.Items {
+				o.AddLabels(tt.inputLabels)
+				if len(tt.expectedLabels) != len(o.object.GetLabels()) {
+					t.Errorf("Expected length of labels to be %v but is %v", len(tt.expectedLabels), len(o.object.GetLabels()))
+				}
+				for k, v := range tt.expectedLabels {
+					if o.object.GetLabels()[k] != v {
+						t.Fatalf("unexpected result, expected ========\n%v\n\nactual ========\n%v\n", tt.expectedLabels, o.object.GetLabels())
```
Comment (justinsb) [inline pkg/patterns/declarative/pkg/manifest/objects_test.go:215]: cmp.Diff produces really nice output in this case (`"github.com/google/go-cmp/cmp"`)

**Comment by justinsb** [comment]
Thanks @atoato88 - great to get more test coverage here !

/approve
/lgtm

**Comment by justinsb** [comment]
/retest

Test failure was unrelated, should have been fixed by #155

**Pull request merged**